Branch-SDK/src/io/branch/referral/BranchException.java

```diff
@@ -0,0 +1,49 @@
+package io.branch.referral;
+
+/**
+ * <p>
+ * Class for representing any Branch SDK exceptions
+ * Exception details are provided in exception message.
+ * </p>
+ */
+public class BranchException extends Exception {
```

Comment:
Mmm, after a second thought, I agree that it makes sense to have both error and exception classes.